File: apps/mobile/lib/domain/usecases/grade/delete_grade_usecase.dart
import '../../../core/error/failures.dart';
import '../../../core/usecases/usecase.dart';
import '../../../core/utils/either.dart';
import '../../repositories/grade/grade_repository.dart';

class DeleteGradeUseCase implements UseCase<Either<Failure, void>, String> {
  final GradeRepository repository;

  DeleteGradeUseCase(this.repository);

  @override
  Future<Either<Failure, void>> call(String params) async {
    return await repository.deleteGrade(params);
  }
}

File: services/user_service/src/domain/usecases/grade/DeleteGradeUseCase.ts
import { IGradeRepository } from "../../repositories/IGradeRepository";

export class DeleteGradeUseCase {
  constructor(private gradeRepository: IGradeRepository) {}

  async execute(id: string): Promise<boolean> {
    const grade = await this.gradeRepository.findById(id);
    if (!grade) {
      throw new Error("Grade not found");
    }

    return await this.gradeRepository.delete(id);
  }
}
